src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.Emitter.cs

```diff
@@ -630,38 +666,118 @@ private void EmitObjectInit(TypeSpec type, string expressionForMemberAccess, Ini
                 {
                     EmitAssignment(expressionForMemberAccess, expressionForInit);
                 }
+
+                string GetDisplayString(TypeSpec type) =>
+                    initKind is InitializationKind.DeclarationFromRootMethod
+                    ? type.FullyQualifiedDisplayString
+                    : type.MinimalDisplayString;
             }
 
-            private void EmitCastToIConfigurationSection()
+            private void EmitIConfigurationHasValueCheck(MethodSpecifier method)
             {
-                _writer.WriteBlockStart($"if ({Literal.configuration} is not {TypeFullName.IConfigurationSection} {Literal.section})");
-                _writer.WriteLine("throw new global::System.InvalidOperationException();");
-                _writer.WriteBlockEnd();
+                string returnStatement = method is MethodSpecifier.Configure
+                    ? "return;"
+                    : "return default;";
+
+                _writer.WriteBlock($$"""
+if (!{{FullyQualifiedDisplayName.Helpers}}.{{Identifier.HasValue}}({{Identifier.configuration}}))
+{
+    {{returnStatement}}
+}
+""");
```

Comment:
Nit: you can indent the closing `"""` without changing the whitespace values of the literal. This would make code easier to read.

Reply:
How many levels of indentation are you thinking? I'm not sure how it would make the code block more readable.

Reply:
I don't think so. In my experience one would have to scroll up/down to find the indentation of the wrapping logic then adjust accordingly. If there are changes to the wrapping indentation, the nested source would have to adjust as well. I think it's easier if the writer handles the indentation so that nested source can be written agnostically.

Reply:
I understand your feedback better. I stuck to the thought that the triple slash feature would also capture indentation but that's not the case as you pointed out. Per private conversation, I'll add indentation to the literals to make  them look less jarring.